lib/Sema/CSBindings.cpp

```diff
@@ -229,8 +229,8 @@ ConstraintSystem::getPotentialBindings(TypeVariableType *typeVar) {
 
     case ConstraintKind::Defaultable:
       // Do these in a separate pass.
-      if (getFixedTypeRecursive(constraint->getFirstType(), true)
-              ->getAs<TypeVariableType>() == typeVar) {
+      if (simplifyType(constraint->getFirstType())->getAs<TypeVariableType>() ==
```

Comment:
Is this part necessary?

Reply:
It looks like if I make `getFixedTypeRecursive` actually return representative it crashes trying to apply arguments in `applySolution`. I'm going to take a closer look at that, so maybe `simplifyType` is not actually required here, but it seems like `getFixedTypeRecursive` was not intended to return representative type variable...